packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1316,6 +1316,35 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
     return Size.zero;
   }
 
+  @override
+  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
+    // TODO: needs a more concise way to get the input semantics node.
+    final SemanticsNode input = children.firstWhere((SemanticsNode element) => element.hasFlag(SemanticsFlag.isTextField));
+    input
+      ..rect = node.rect
+      ..transform = Matrix4.identity();
+    super.assembleSemanticsNode(node, config, children);
+  }
+
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    config.childSemanticsConfigMutator = (Map<AbstractNode, List<SemanticsConfiguration>> childConfigs) {
```

Comment:
Would something like this work: a callback that gives you access to your render children and the associated SemanticConfig and you return the configs that you want to have merged into the parent. The remaining one stay explicit.